bench.js

```diff
@@ -119,4 +120,108 @@ async function benchBranch(app, config) {
     }
 }
 
-module.exports = benchBranch;
\ No newline at end of file
+var RuntimeBenchmarkConfigs = {
+    "pallet": {
+        title: "Runtime Benchmarks Pallet",
+        branchCommand: 'cargo run --release --features runtime-benchmarks --manifest-path bin/node/cli/Cargo.toml -- benchmark --chain dev --steps 50 --repeat 20 --extrinsic "*" --raw --execution=wasm --wasm-execution=compiled --output --pallet',
+    },
+    "custom": {
+        title: "Runtime Benchmarks Custom",
+        branchCommand: 'cargo run --release --features runtime-benchmarks --manifest-path bin/node/cli/Cargo.toml -- benchmark',
+    }
+}
+
+function checkRuntimeBenchmarkCommand(command) {
+    let required = ["benchmark", "--pallet", "--extrinsic", "--execution", "--wasm-execution", "--steps", "--repeat", "--chain"];
+    let missing = [];
+    for (const flag of required) {
+        if (!command.includes(flag)) {
+            missing.push(flag);
+        }
+    }
+
+    return missing;
+}
+
+async function benchmarkRuntime(app, config) {
+    app.log("Waiting our turn to run benchmark...")
+
+    const release = await mutex.acquire();
+
+    try {
+        if (config.extra.split(" ").length < 2) {
+            return errorResult(`Incomplete command.`)
+        }
+
+        let command = config.extra.split(" ")[0];
+        var benchConfig = RuntimeBenchmarkConfigs[command];
+        var extra = config.extra.split(" ").slice(1).join(" ").trim();
```

Comment:
this allows arbitrary shell command execution on our bench machine, by the means of posting a comment 

combining with the fact that we want to push branch from it, not very secure 🤔

Reply:
need at least strip all "#&|;:" symbols from there

Reply:
done